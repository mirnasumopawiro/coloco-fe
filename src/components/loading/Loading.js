import React from "react";
import Loader from "react-loader-spinner";
import PropTypes from "prop-types";
import { Modal, ModalBody } from "reactstrap";
import { CModal, CModalBody } from "@coreui/react";

const Loading = (props) => {
  return (
    <CModal
      {...props}
      style={{
        background: "transparent",
        border: "none",
        alignItems: "center",
      }}
      size="sm"
      backdrop
      fade
      className="loading-wrapper"
    >
      <CModalBody style={{}}>
        <Loader type="TailSpin" color="white" height="100" width="100" />
      </CModalBody>
    </CModal>
  );
};

Loading.propTypes = {
  isOpen: PropTypes.bool,
};

export default Loading;
