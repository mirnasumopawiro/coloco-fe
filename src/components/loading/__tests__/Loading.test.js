import React from 'react';
import {shallow} from 'enzyme';
import Loading from '../Loading';

const props = {
  isOpen: true
};

test('Should render Loading without crashing', () => {
  const tree = shallow(<Loading {...props}/>);
  expect(tree).toMatchSnapshot();
});
