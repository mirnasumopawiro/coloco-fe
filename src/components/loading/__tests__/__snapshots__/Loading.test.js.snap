// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Loading without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    isOpen={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": true,
      "backdrop": true,
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": <ModalBody
        tag="div"
      >
        <Loader
          color="#00BFFF"
          height="75"
          type="ThreeDots"
          visible={true}
          width="75"
        />
      </ModalBody>,
      "className": "loading-wrapper",
      "fade": true,
      "isOpen": true,
      "keyboard": true,
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "size": "sm",
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Loader
          color="#00BFFF"
          height="75"
          type="ThreeDots"
          visible={true}
          width="75"
        />,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#00BFFF",
          "height": "75",
          "type": "ThreeDots",
          "visible": true,
          "width": "75",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": <ModalBody
          tag="div"
        >
          <Loader
            color="#00BFFF"
            height="75"
            type="ThreeDots"
            visible={true}
            width="75"
          />
        </ModalBody>,
        "className": "loading-wrapper",
        "fade": true,
        "isOpen": true,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "size": "sm",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Loader
            color="#00BFFF"
            height="75"
            type="ThreeDots"
            visible={true}
            width="75"
          />,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#00BFFF",
            "height": "75",
            "type": "ThreeDots",
            "visible": true,
            "width": "75",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
